drivers/modem/hl7800.c

```diff
@@ -347,6 +348,14 @@ static const char TIME_STRING_FORMAT[] = "\"yy/MM/dd,hh:mm:ss?zz\"";
 		}                                                              \
 	} while (0)
 
+#define SEND_AT_CMD_ARGS_EXPECT_OK(c, t, n)                                    \
+	do {                                                                   \
+		ret = send_at_cmd(NULL, (c), (t), (n), false);                 \
+		if (ret < 0) {                                                 \
+			LOG_ERR("%s result:%d", (c), ret);                     \
+		}                                                              \
+	} while (0)
```

Comment:
I wonder if we could do this slightly differently like this
```
#define SEND_AT_CMD_ARGS_EXPECT_OK(c, t, n) \
({                                                                      \
	ret = send_at_cmd(NULL, (c), (t), (n), false);                 \
	if (ret < 0) {                                                 \
		LOG_ERR("%s result:%d", (c), ret);                     \
	}                                                                 \
        ret;                                                             \
})
```
This way we could do
```
ret = SEND_AT_CMD_ARGS_EXPECT_OK("AT+KSRAT=0", MDM_CMD_KSRAT_TIMEOUT, 0);
if (ret < 0) {
	goto error;
}
```
which would not "hide" the ret value. If one reads the patch, it is not obvious where the value of ret comes from because it is hidden inside the macro.